### ISSUE TITLE
HTML (altchunk) not supported/displayed when opening in LibreOffice

### ISSUE THREAD

**Issue by pawepaw** [guigrpa/docx-templates#201]
Wanted to use docx templates in our process to generate a pdf report. Just spotted that LibreOffice isn't displaying that HTML rendered with docx-templates `altchunk` at all.

Is it only me who is experiencing such problems? Is there any other option to render HTML that's gonna be opened by LibreOffice?

**Comment by jjhbw** [comment]
I expect this to be a LibreOffice thing. Leaving this open for others to comment.

**Comment by pawepaw** [comment]
yes that's totaly LibreOffice thing. Just wanted to ask if you're aware of any workaround.

**Comment by pawepaw** [comment]
i.e i see that https://docxtemplater.com/demo/#html is handling that perfectly. Embedded some HTML table and libre office displays it properly. hmmmm

**Comment by jjhbw** [comment]
So you mean that it should be possible? If you can find what exactly it does differently (i.e. does the HTML look different?)  we may be able to build a patch.

**Comment by pawepaw** [comment]
well I'm not sure if it should be possible, not an expert on that field. I've checked docxtemplater result and it seems that they are generating openxml based on provided HTML, so it's basically different approach.

**Comment by veich** [comment]
HTML is not generated for me as well - like it "eats" all the HTML and all I get nothing in the places where HTML should have been.
I'm using version `"docx-templates": "^4.5.4"`

**Comment by veich** [comment]
> well I'm not sure if it should be possible, not an expert on that field. I've checked docxtemplater result and it seems that they are generating openxml based on provided HTML, so it's basically different approach.

what is the relationship between `docxtemplater` (which is commercial product) and this lib?

**Comment by jjhbw** [comment]
> what is the relationship between docxtemplater (which is commercial product) and this lib?

There is none. Just a slightly inconvenient naming similarity ;)

**Comment by veich** [comment]
**FYI @all**
partial solution (at least the one I opted for)
docx is basically .zip-ed xml so my plan is to directly edit xml so I don't have to use libraries (except some for XML manipulation but that is pretty straightforward)

**Comment by qngdt** [comment]
> **FYI @ALL**
> partial solution (at least the one I opted for)
> docx is basically .zip-ed xml so my plan is to directly edit xml so I don't have to use libraries (except some for XML manipulation but that is pretty straightforward)

Could you be more specific about your approach? I'm having the same issue.

**Comment by veich** [comment]
> Could you be more specific about your approach? I'm having the same issue.

unzip the .docx file and go to `word/document.xml` you can view and edit the file and that's it
look how current structure looks like and just imitate the elements you wish to add and that's it

**Comment by carlos-peru** [comment]
I have the same issue. What I noticed is that LibreOffice can properly display html directly. In addition, if I open a html file (with html content) and save it as a docx, it still displays the content correctly. The problem seems to be the embedding of html in this library that it only renders properly in MS Word. It does not render on Libre Office, Open Office or Google Docs. At least in my case working on Mac.

**Comment by carlos-peru** [comment]
So apparently this is due to lack of support for altchunk as it seems to be non-standard. Even MS Word re-formats/re-parses the files with altchunk if you "save as" the docx.

**Comment by Hariprasath-G** [comment]
Please note that until a document that contains altChunk elements is opened and saved in Office, it still contains the altChunk parts, and not normal WordprocessingML markup of paragraphs, runs, and text elements.  The solution with SharePoint 2010 is that you can use Word Automation Services to update the documents that contain altChunk elements.  After Word Automation Services processes it, the document will contain paragraphs, runs, and text elements.

https://ankushbhatia.wordpress.com/2010/10/22/altchunk-an-interesting-scenario/?unapproved=4594&moderation-hash=4178e78d24528e1c1c061e585d38bdc8#comment-4594

**Comment by FelixRelli** [comment]
Instead of relying on altchunks it could be done similar to what https://www.npmjs.com/package/html-to-docx does

**Comment by Akbar0102** [comment]
Is there a solution for this? I have been using it to generate DOCX files and then convert them to PDF. Everything worked well on Windows, but when I moved to Linux and started using LibreOffice, the generated DOCX files do not display content from HTML. This also affects the generated PDF files.

**Comment by jmls** [comment]
hitting the same problem :(

**Comment by alexrosepizant** [comment]
I created a lib to handle html tags and replace them by docx tags, if you want to take a look: https://github.com/alexrosepizant/docx-templates-html

**Comment by alexrosepizant** [comment]
> > Instead of relying on altchunks it could be done similar to what https://www.npmjs.com/package/html-to-docx does.
> 
> Is there any way to export xml via https://www.npmjs.com/package/html-to-docx and then import the structure into the document by inserting xml? like: +++INS text+++ (https://github.com/guigrpa/docx-templates?tab=readme-ov-file#inserting-literal-xml)

I created a lib to handle html tags and replace them by docx tags, if you want to take a look: https://github.com/alexrosepizant/docx-templates-html

**Comment by alexrosepizant** [comment]
I write a library to improve html rendering in docx templates if you want to check: https://github.com/alexrosepizant/docx-templates-html
